libp2p/builders.nim

```diff
@@ -282,8 +283,11 @@ proc withAutonat*(b: SwitchBuilder): SwitchBuilder =
   b.autonat = true
   b
 
-proc withAutonatV2*(b: SwitchBuilder): SwitchBuilder =
+proc withAutonatV2*(
+    b: SwitchBuilder, config: AutonatV2Config = AutonatV2Config.new()
```

Comment:
Consider making the config optional, so autonatV2 can work with sane defaults and dev does not have to worry about passing a config

Reply:
So we either have it inside an `Opt` or not but with defaults. I prefer the latter to avoid the opt checks every time.